test/runtests.jl

```diff
@@ -24,8 +24,18 @@ end
     @test setproperties(o, (a=2, b=3.0)) === AB(2,3.0)
     @test setproperties(o, a=2, b=3.0) === AB(2,3.0)
 
-    @test_throws ArgumentError setproperties(o, (a=2, c=3.0))
-    @test_throws ArgumentError setproperties(o, a=2, c=3.0)
+    res = @test_throws ArgumentError setproperties(o, (a=2, this_field_does_not_exist=3.0))
+    msg = sprint(showerror, res.value)
```

Comment:
Is accessing the error message like this is considered a public API?

Reply:
I asked to make it so a while ago https://github.com/JuliaLang/julia/pull/31854